src/averageAges.js

```diff
@@ -55,10 +71,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredChildren = people.filter(person => onlyWithSon
+    ? people.some(mother => mother.name === person.mother
+      && person.sex === 'm')
+    : people.find(mother => mother.name === person.mother)
+  );
```

Comment:
Now everything is written correctly, but such code is difficult to read and maintain. Break it into smaller parts.